packages/eslint-plugin/lib/config/by-project-type/monorepo-workspace.ts

```diff
@@ -1,22 +1,28 @@
 import type { Linter } from "eslint";
+import { concat } from "eslint-flat-config-utils";
+import core from "../core.ts";
+import jest from "../jest.ts";
+import mdx from "../mdx.ts";
+import packageJson from "../package-json.ts";
+import testingLibrary from "../testing-library.ts";
+import typescript from "../typescript.ts";
 
-const config: Linter.Config = {
-    plugins: ["@workleap"],
-    extends: [
-        "plugin:@workleap/core",
-        "plugin:@workleap/typescript",
-        "plugin:@workleap/jest",
-        "plugin:@workleap/testing-library",
-        "plugin:@workleap/mdx",
-        "plugin:@workleap/package-json",
-        "plugin:@workleap/yaml"
-    ],
-    rules: {
-        "package-json/valid-version": "off"
+const config: Linter.FlatConfig[] = await concat(
```

Comment:
I am wondering about the "await".

In the library documentation he is not awaiting the concat:

```js
import { concat } from 'eslint-flat-config-utils'

export default concat(
  {
    plugins: {},
    rules: {},
  },
  // It can also takes a array of configs:
  [
    {
      plugins: {},
      rules: {},
    }
    // ...
  ],
  // Or promises:
  Promise.resolve({
    files: ['*.ts'],
    rules: {},
  })
  // ...
)
```

Reply:
`concat` always returns a promise, even if none of the things passed in are promises. So the issue I was struggling with was when exactly to handle this promise. ESLint can actually take a promise that resolves to a config array (so it can take the direct output of `concat`), but it can't take an array of promises.

If I don't await at the top of each config module, then each one would export a promise. This forces any consumer to use either `concat` or `Promise.All` + some array spread to resolve all of these before it goes to ESLint. And it gets a little more difficult if you need to overwrite values in your config, because you'd need to resolve the promise first and then map or reduce that into the config you want. That all adds just a little more complexity for setup than a typical ESLint config would require.

Top level await there seemed like the best solution.